java/test/jmri/jmrit/display/layoutEditor/LayoutEditorTest.java

```diff
@@ -919,7 +919,7 @@ public void testToolBarPositionLeft() {
                 + Bundle.getMessage("ToolBarSide") + "/"
                 + Bundle.getMessage("ToolBarSideLeft"), "/");
 
-        delay(500); //TODO: switch to JemmyUtil.delay
+        JemmyUtil.delay(500); //TODO: switch to JemmyUtil.delay
```

Comment:
Make sure you remove the todo comments from these lines.

Also, I am a bit concerned that you have a wrapper around a sleep here.  It should be sufficient for any graphical operation to use the Jemmy QueueTool to wait for the awt event queue to be empty.  Raw half second delays in tests are a code smell.

Reply:
That's great in theory… in practice however I'm finding that sometimes neither QueueTool().waitEmpty() nor EventTool().waitNoEvent() work… the 1/2 second was somewhat arbitrary. "Code smell" is an excellent description of Jemmy. I'm finding it woefully under documented. For example: what's the difference between .push() and .doClick()? In JennyUtils three of the four enterClickAndLeave methods used .doClick() and the other .push(). Why? Changing them all to use .push broke the tests so I changed them all to .doClick() (which only makes sense given the name of the methods). But when to use one vs. the other? Not a clue.

Reply:
doClick and push cause different events to be sent to the event queue.  Some events are triggered by a mouse down event ( which both send ) some events are triggered by a mouse up event ( doClick sends that ).  doClick is roughly the equivalent of a push followed by a release.

Where you really have to be careful with this is that some actions don’t respond correctly if the button has already been released when the event is handled ( I.e. if too little time occurs between the mouse up and mouse down ).  The use of push in the JemmyUtils was intentional where it is used.

Reply:
There's a setting for the time between the press and the release… I suspect it's a global setting so if it's changed it should be changed back so that it won't effect other tests.

Reply:
All delay/sleep code has been REMOVED.